source/service.go

```diff
@@ -732,7 +732,7 @@ func (sc *serviceSource) extractNodePortEndpoints(svc *v1.Service, hostname stri
 				protocol = "tcp"
 			}
 
-			recordName := fmt.Sprintf("_%s._%s.%s", serviceName, protocol, hostname)
+			recordName := fmt.Sprintf("_%s._%s.%s.", serviceName, protocol, hostname)
```

Comment:
Same 

```
recordName := fmt.Sprintf("_%s._%s.%s.", serviceName, protocol, hostname, provider.EnsureTrailingDot(hostname))
```

less effiicient, but this will make it simpler to identify potential targets for refactoring, improvements